packages/flutter/lib/src/painting/box_decoration.dart

```diff
@@ -321,9 +321,26 @@ class _BoxDecorationPainter extends BoxPainter {
   ImageInfo _image;
 
   void _paintBackgroundImage(Canvas canvas, Rect rect, ImageConfiguration configuration) {
+    // TODO(ianh): factor this out into a DecorationImage.paint method.
     final DecorationImage backgroundImage = _decoration.image;
     if (backgroundImage == null)
       return;
+
+    bool flipHorizontally = false;
+    if (backgroundImage.matchTextDirection) {
+      // We check this first so that the assert will fire immediately, not just when the
+      // image is ready.
+      assert(configuration.textDirection != null, 'matchTextDirection can only be used when a TextDirection is available.');
+      switch (configuration.textDirection) {
+        case TextDirection.rtl:
+          flipHorizontally = true;
+          break;
+        case TextDirection.ltr:
+          // false is the default
+          break;
+      }
+    }
```

Comment:
I'd factor this out into a `textDirectionIsReversed` analogous to `axisDirectionIsReversed`.  Then this becomes:

```
bool flipHorizontally = !backgroundImage.matchTextDirection || textDirectionIsReversed(configuration.textDirection);
```

Reply:
I take it back.  We don't want other clients to call `textDirectionIsReversed`.